### PR TITLE
Snowbridge: Refactor inbound message conversion

### PR DESCRIPTION
**Pull request by yrong**
# Description

- The current `MessageToXcm` inbound converter is somewhat verbose, with injected types that are sparse and unorganized, This change cleans up and refactors the code accordingly. (I can’t find the original comment, but I vaguely recall it being mentioned by @acatangiu  in a previous PR review).

- Add more tests cover some edge cases, such as claiming assets trapped on AH from a parachain.

### CODE REVIEW

File: prdoc/pr_9510.prdoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+title: "Snowbridge: Refactor inbound message conversion"
+doc:
+- audience: Runtime Dev
+  description: |-
+    Refactor MessageToXcm inbound converter to make it more compact and organized, 
+    update runtime configuration accordingly
+crates:
+- name: snowbridge-inbound-queue-primitives
+  bump: minor
```
Comment (acatangiu) [inline prdoc/pr_9510.prdoc]: this is breaking change, needs to be major bump

```suggestion
  bump: major
```

**Review by acatangiu — APPROVED** [review]
looks good!

**Review by claravanstaden — APPROVED** [review]
Nice! 🧹
